packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -182,10 +182,245 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
+
+  /**
+   * Permits all DynamoDB operations ("dynamodb:*") to an IAM principal.
+   * @param grantee The principal to grant access to
+   */
+  grantFullAccess(grantee: iam.IGrantable): iam.Grant;
+}
+
+/**
+ * Reference to a dynamodb table.
+ */
+export interface TableAttributes {
+  /**
+   * The ARN of the dynamodb table.
+   *
+   * @default no table arn
+   */
+  readonly tableArn?: string;
+
+  /**
+   * The table name of the dynamodb table.
+   *
+   * @default no table name
+   */
+  readonly tableName?: string;
+
+  /**
+   * The stream arn of the dynamodb.
+   *
+   * @default no table stream arn
+   */
+  readonly tableStreamArn?: string;
+
+  /**
+   * Indicate if the table has global or local secondary indexes.
+   */
+  readonly hasIndex: boolean;
+}
+abstract class TableBase extends Resource implements ITable {
+  /**
+   * @attribute
+   */
+  public abstract readonly tableArn: string;
+
+  /**
+   * @attribute
+   */
+  public abstract readonly tableName: string;
+
+  /**
+   * @attribute
+   */
+  public abstract readonly tableStreamArn: string | undefined;
+
+  /**
+   * Adds an IAM policy statement associated with this table to an IAM
+   * principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:PutItem", "dynamodb:GetItem", ...)
+   */
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+        Lazy.stringValue({ produce: () => this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE })
+      ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Adds an IAM policy statement associated with this table's stream to an
+   * IAM principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:DescribeStream", "dynamodb:GetRecords", ...)
+   */
+  public grantStream(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    if (!this.tableStreamArn) {
+      throw new Error(`DynamoDB Streams must be enabled on the table ${this.node.path}`);
+    }
+
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [this.tableStreamArn],
+      scope: this,
+    });
+  }
+
+  /**
+   * Permits an IAM principal all data read operations from this table:
+   * BatchGetItem, GetRecords, GetShardIterator, Query, GetItem, Scan.
+   * @param grantee The principal to grant access to
+   */
+  public grantReadData(grantee: iam.IGrantable) {
+    return this.grant(grantee, ...READ_DATA_ACTIONS);
+  }
+
+  /**
+   * Permits an IAM Principal to list streams attached to current dynamodb table.
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  public grantTableListStreams(grantee: iam.IGrantable): iam.Grant {
+    if (!this.tableStreamArn) {
```

Comment:
I wonder if this validation makes sense for imported Tables... maybe it makes sense to get rid of it in that case?